tests/MyEventEmitter.test.js

```diff
@@ -0,0 +1,471 @@
+/* eslint-disable max-len */
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const MyEventEmitter = require('../src/MyEventEmitter');
+const EventEmitter = require('events');
+
+const getRandomEventName = () => `testEvent-${Math.random()}`;
+
+const getRandomArgument = () => {
+  const types = ['string', 'number', 'boolean', 'object', 'undefined', 'null'];
+  const type = Math.floor(Math.random() * types.length);
+
+  switch (type) {
+    case 0:
+      return 'test';
+    case 1:
+      return Math.random();
+    case 2:
+      return Math.random() > 0.5;
+    case 3:
+      return {
+        test: 'test',
+        test2: Math.random(),
+        test3: Math.random() > 0.5,
+      };
+    case 4:
+      return undefined;
+    case 5:
+      return null;
+    default:
+      return undefined;
+  }
+};
+
+const getRandomArguments = () => {
+  const count = Math.floor(Math.random() * 10) + 1;
+
+  return Array.from({ length: count }, getRandomArgument);
+};
+
+describe('MyEventEmitter', () => {
+  let emitter;
+
+  beforeEach(() => {
+    emitter = new MyEventEmitter();
+  });
+
+  test('should NOT use external library dependencies', () => {
+    const sourceFile = path
+      .resolve(__dirname, '..', 'src', 'MyEventEmitter.js');
+    const fileContent = fs.readFileSync(sourceFile, 'utf8');
+    const hasCommonJSRequire = /require\(/g.test(fileContent);
+    const hasESImports = /import\s+\S+\s+from\s+['"].+['"]/g.test(fileContent);
+
+    expect(hasCommonJSRequire).toBe(false);
+    expect(hasESImports).toBe(false);
+  });
+
+  test('should NOT extend from Node.js EventEmitter', () => {
+    expect(emitter).not.toBeInstanceOf(EventEmitter);
+  });
+
+  describe('"listenerCount" method', () => {
+    test('should always return 0 for an event without listeners', () => {
+      expect(emitter.listenerCount(getRandomEventName()))
```

Comment:
Just hardcode 2 names here 
later you will add those listener and check if count for 1 of them became 1 and stayed 0 for the other

Don't add random test data, it is really hard to debug
We are not trying to stop cheater from cheating

Reply:
I am sure that using random data is the best practice. This will help our students understand programming and testing concepts better.